modules/geolocation/geolocation.py

```diff
@@ -231,20 +232,33 @@ def get_non_collinear_points(self, coordinatesArray):
             for j in range(0, NUM_POINTS_NEEDED):
                 points[j] = coordinatesArray[(i + j) % len(coordinatesArray)]
 
-            # Check collinearity of all possible combinations
             areNotFourCollinear = True
-            for i in range(0, NUM_POINTS_NEEDED):
-                areNotFourCollinear &= not self.__are_three_points_collinear(points[i],
-                                                                             points[(i + 1) % NUM_POINTS_NEEDED],
-                                                                             points[(i + 2) % NUM_POINTS_NEEDED])
+
+            # Check collinearity of all possible combinations
+            for k in range(0, NUM_POINTS_NEEDED):
+                areNotFourCollinear &= not self.__are_three_points_collinear(points[k],
+                                                                             points[(k + 1) % NUM_POINTS_NEEDED],
+                                                                             points[(k + 2) % NUM_POINTS_NEEDED])
+                # Store indexes of current iteration
+                indexes[0] = (i)
+                indexes[1] = ((i + 1) % len(coordinatesArray))
+                indexes[2] = ((i + 2) % len(coordinatesArray))
+                indexes[3] = ((i + 3) % len(coordinatesArray))
```

Comment:
Move out of the k loop. Can be turned into its own loop (actually, you can make the j loop do double-duty).